src/gsy_e/models/area/__init__.py

```diff
@@ -646,3 +672,17 @@ def is_market_balancing(self, market_id: str) -> bool:
     def is_market_future(self, market_id: str) -> bool:
         """Return True if market_id belongs to a FUTURE market."""
         return market_id == self.future_markets.id
+
+
+class Market(Area):
+    """Class to define geographical market areas that can contain children (areas or assets)."""
+
+
+class Asset(Area):
+    """Class to define assets (devices). These instances cannot contain children."""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        if self.children:
+            raise ValueError(f"{self.__class__.__name__} instances can't have children.")
```

Comment:
This is the core part of the renaming. We create a new `Market` class that has the same functionality as the current `Area` class. Moreover, the `Asset` class is basically the same but it can't be instantiated with children.

Note: in theory, it's possible to create an instance of `Asset` without children and later assign them with `asset.childer = ...`. I tried to create setters for it but it would get messy. If you want I can work on it, but it'll need some refactoring.

Reply:
Leaving it like you propose works for me. If a user tries to add a children to an Asset, it is his/her own fold IMO.
One minor thing though, how about first checking for the children and only then call the `super().__init__()` ?

Reply:
> One minor thing though, how about first checking for the children and only then call the `super().__init__()` ?
Can you please do this? IMO this saves the "effort" of initialisation of the super class if the user accidentally provided a list of children.

Reply:
It was discussed offline that it only makes sense now. Basically: `self.children` is not populated  before the `super().__init__()`  is called.